features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/openbmp/BmpKafkaProducer.java

```diff
@@ -59,7 +59,18 @@ public BmpKafkaProducer(final String topicPrefix,
     }
 
     private static KafkaProducer<String, String> buildProducer(final Map<String, Object> kafkaConfig) {
-        // TODO fooker: Apply defaults (steal from https://github.com/SNAS/openbmp/blob/1a615a3c75a0143cc87ec70458471f0af67d3929/Server/src/kafka/MsgBusImpl_kafka.cpp#L162) (see https://issues.opennms.org/browse/NMS-12574)
+        // Apply some defaults
+        kafkaConfig.putIfAbsent("batch.num.messages", 100);
+        kafkaConfig.putIfAbsent("compression.codec", "snappy");
+        kafkaConfig.putIfAbsent("message.max.bytes", 1000000);
+        kafkaConfig.putIfAbsent("receive.message.max.bytes", 100000000);
```

Comment:
Can't find this property with quick search on producer config.  If you can directly get them from kafka client as Jesse suggested, that would rule out any outdated configuration.